sbin/root-mount

```diff
@@ -136,6 +139,15 @@ factory_reset_if_needed()
 		if factory-reset-lvm $LVM_ROOT_SIZE $LVM_RESERVED_MB; then
 			# Successful recovery.
 
+			if test "$CLEAR_EXTERNAL_MEDIA" = "1"; then
+				CLEAR_EXTERNAL_COMMAND=`which factory-reset-external`
+				if ! test "$CLEAR_EXTERNAL_COMMAND" != ""; then
```

Comment:
not not equal... how about

    if test "$CLEAR_EXTERNAL_COMMAND" == ""; then

or

    if test -z "$CLEAR_EXTERNAL_COMMAND";  then